src/main/java/gov/cabinetofice/gapuserservice/service/OneLoginService.java

```diff
@@ -282,14 +297,36 @@ public void migrateUser(final User user, final String jwt) {
                 .block();
     }
 
+    public HttpResponse logoutUser(final Cookie customJWTCookie, final HttpServletResponse response) {
+        try {
+
+            final DecodedJWT decodedJwt = customJwtService.decodedJwt(customJWTCookie.getValue());
+            final JwtPayload payload = customJwtService.decodeTheTokenPayloadInAReadableFormat(decodedJwt);
+
+            HttpResponse res = RestUtils.getRequest(oneLoginLogoutEndpoint + "?id_token_hint=" + sanitizeCode(payload.getIdToken()) +
+                    "&post_logout_redirect_uri=" + postLogoutRedirectUri);
+
+            if(res.getStatusLine().getStatusCode() == 401 || res.getStatusLine().getStatusCode() == 403) {
+                final Logger logger = null;
```

Comment:
You don't need to initialise a logger here.

You can use @Log4j2 above the class and then reference log.error(....)